MedallionShell/Shell.cs

```diff
@@ -76,6 +83,24 @@ public Command Run(string executable, IEnumerable<object>? arguments = null, Act
             return command;
         }
 
+        // internal for testing
+        internal static string? GetFullPathUsingSystemPathOrDefault(string executable)
+        {
+            var paths = Environment.GetEnvironmentVariable("PATH")!.Split(Path.PathSeparator);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var pathExtensions = Environment.GetEnvironmentVariable("PATHEXT")!
```

Comment:
On Windows, both `dotnet.exe --version` and `dotnet --version` work, but most users just type `dotnet`. Therefore, I think we should allow users to not pass in the extension.

Reply:
Actually this is the default behavior.